### PR TITLE
Feature navbar

### PR DESCRIPTION
**Pull request by amanbayev**
#### Updated 21.11.2017 (followed suggested comments - thank you for feedback!)

- [x] removed JS
- [x] moved css to less
- [x] added home icon suggested in comments
- [x] restored original template md file contents
- [x] added original version (alternate styling) of navbar without buttons, but with flat links
- [x] removed custom classes
- [x] removed button version, went with flat one
- [x] removed home icon - i think creating matching iconset is another subtask of this project
- [x] minifed example html and less

### Navbar element for PaperCSS
Added a small paper-navbar element which does good job at creating simple mobile-friendly top navbar

This is what it looks like in HTML:
```html
<nav>
      <label for="menu-toggle">&#9776;</label>
      <input type="checkbox" id="menu-toggle" role="button"></input>
      <ul>
        <li class="active"><a href="#">Home</a></li>
        <li><a href="#">About</a></li>
        <li><a href="#">Portfolio</a></li>
        <li><a href="#">News</a></li>
        <li><a href="#">Contact</a></li>
      </ul>
</nav>
```

The less file is located in navbar.less:

```less
@grey_background: #c2c2c2;
.active {
	background: @grey_background;
}
nav {
	display: block;
	margin-top: 0px;
	margin-left: 0px;
	label {
		display: none;
	}
	input {
		display: none;
		&:checked {
			&~ul {
				display: block;
			}
		}
	}
	ul {
		&:not(:first-child) {
			margin-top: -20px;
		}
	}
	li {
		font-style: normal;
		padding-left: 16px;
		padding-right: 16px;
		padding-top: 6px;
		padding-bottom: 6px;
		display: inline-block;
		margin-right: 8px;
		&:hover {
			background: @grey_background;
		}
		&:before {
			content: "";
		}
		a {
			background-image: none;
		}
	}
}
@media screen and (max-width : 760px) {
	nav {
		ul {
			position: static;
			&:not(:first-child) {
				display: none;
			}
			li {
				&:not(:first-child) {
					margin-top: 30px;
				}
				float: none;
	      display: block;
	      text-align: left;
				margin-bottom: 1px;
			}
		}
		label {
			display: block;
		}
	}
}
```

## Developer Certificate of Origin

- [x] I certify that these changes according to the Developer Certificate of Origin 1.1 as described at <https://developercertificate.org/>.

## Sample pictures

- [x] removed pictures

## Further details

This is just a start. Perhaps someone can help enhance it.
I think having a simple way to add navbar would be awesome.

p.s.: I love PaperCSS so far - great framework

### CODE REVIEW

**Comment by afzalsayed96** [comment]
Navbar is a good idea. Can you add screenshots? The sample pictures seem to be not working

**Comment by amanbayev** [comment]
I was thinking something simple to keep the style simple.
But for future would be nice to do the Sidebar as well. So we can feature
fully PaperCSS wiki and admin dashboard examples (like in getbootstrap
docs) built using the PaperCSS itself

Talgat Amanbayev
amanbayev@gmail.com
t.amanbayev@gmail.com

2017-11-20 12:42 GMT+06:00 afzalsayed96 <notifications@github.com>:

File: index.html [hunk 1]
```diff
@@ -1102,6 +1130,17 @@ <h2 id="about">About</h2>
       <p>Made with 💛 by <a href="https://vlaservi.ch" target="_blank">Rhyne</a> and some <a href="https://github.com/rhyneav/papercss/graphs/contributors">fantastic contributors</a>!</p>
     </div>
   </div>
+  <script>
+    /* Toggle between adding and removing the "responsive" class to topnav when the user clicks on the icon */
+    function myFunction() {
```
Comment (Fraham) [inline index.html]: can this function have a better name, "ToggleNavBarVisibility"

File: index.html [hunk 2]
```diff
@@ -1102,6 +1130,17 @@ <h2 id="about">About</h2>
...
+      var x = document.getElementById("myTopnav");
+      if (x.className === "topnav") {
+          x.className += " responsive";
+      } else {
+          x.className = "topnav";
```
Comment (Fraham) [inline index.html]: This wont be nice if the nav bar uses other classes

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,15 +1,113 @@
 ## Brief description
 
-...
+#### Top Nav element for PaperCSS
```
Comment (Fraham) [inline PULL_REQUEST_TEMPLATE.md]: why has this file been edited?

File: demo.css [hunk 4]
```diff
@@ -12,5 +12,61 @@ body {
...
 }
 .summary a {
-    color: #41403e;    
-}
\ No newline at end of file
+    color: #41403e;
+}
+
+/* paper topnav START */
+.topnav {
```
Comment (Fraham) [inline demo.css]: why has the css been included here? You should create a new file in the source folder and add the new nav bar css in there

**Comment by rhyneav** [comment]
Navbar is a great idea! I also like the initial screenshots. 

@Fraham has some great feedback already (thank you, sir). I'd recommend creating a new file, src/navbar.less and importing that into the styles.less file. Also, before changing anything else, make sure you are basing off of the develop branch! 

I would also like to keep JavaScript out of the framework as much as possible. There's a couple of nice Codepens ([like this one](https://codepen.io/andreasstorm/pen/lriga)) that show that it's possible to have a responsive navbar, but without JS.

**Comment by afzalsayed96** [comment]
@amanbayev in topnav2 screenshot maybe a home icon would look nice instead of text
topnav2

**Comment by rhyneav** [comment]
I like that idea! Maybe something like this? 
Home

https://thenounproject.com/term/home/85769/

**Comment by afzalsayed96** [comment]
@rhyneav That's nice! I also tried to create a home icon in SVG format. Check it out.

home icon

File: index.html (hunk 2 above)
Comment (amanbayev) [reply index.html]: I will rewrite it! Thank you for feedback!

File: index.html (hunk 1 above)
Comment (amanbayev) [reply index.html]: Thank you for feedback, I will rewrite it

File: demo.css (hunk 4 above)
Comment (amanbayev) [reply demo.css]: ok, I will add new file. Thank you for feedback!

File: PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (amanbayev) [reply PULL_REQUEST_TEMPLATE.md]: Sorry, it is my first time contributing to an open source via github. I will fix it

**Comment by amanbayev** [comment]
@Fraham 

> Can you change the pull request merge branch so it's pointing to develop instead master
Yes I have changed it, Thank you for feedback!

**Comment by afzalsayed96** [comment]
@amanbayev the navbar looks really clean. Can we also add documentation to it so that others can use this simple paper navbar in their projects?

File: index.html [hunk 5]
```diff
@@ -25,16 +25,31 @@
 </head>
 
 <body>
-  <div class="container paper">
 
+  <div class="container paper">
+    <!-- Topnav START -->
```
Comment (TotomInc) [inline index.html]: Can you please remove comments?

**Comment by afzalsayed96** [comment]
IMO the previous (flat) style for navbar was neat and simple. I miss the links without any border or shadows and equal heights & alignment for nav links, home icon and hamburger icon

**Comment by amanbayev** [comment]
@afzalsayed96 I will create 2 styles I guess - flat and button. little later tonight i hope.

> IMO the previous (flat) style for navbar was neat and simple. I miss the links without any border or shadows and equal heights & alignment for nav links, home icon and hamburger icon

File: index.html (hunk 5 above)
Comment (amanbayev) [reply index.html]: yes, sure. will do.

**Comment by amanbayev** [comment]
@afzalsayed96 I have added original styles for navbar. They seem to work fine. The example usage is given in index.html (commented out section). Please avoid having 2 navbars, the hamburger icon gets ugly when both navbars are positioned to float right and top.

**Comment by amanbayev** [comment]
@TotomInc I will remove comments section to the readme.md I guess (it currently has menu variation example)

> It looks really good, just remove the comments on the index.html file and I think it's ready for a merge!

**Comment by afzalsayed96** [comment]
Thanks @amanbayev !

- There is some overflow issue in the border-button navbar version (check x-axis scroll in mobile view).
image

- Also can you make the flat version the primary navbar. It looks nice and clean with the landing page

- What is the purpose of arrows in links on the nav? Should we remove it?
image

@rhyneav @Fraham @TotomInc what are your thoughts?

**Comment by TotomInc** [comment]
@afzalsayed96 & @amanbayev  I think the purpose of arrows here are certainly for dropdowns. Both an arrow or a [caret](http://fontawesome.io/icon/caret-down/) seems to be OK for the navbar. Otherwise, it looks okay, just need to fix the overflow issue on mobile view. The navbar is compact and clean, fit perfectly PaperCSS theme, great job! :+1:

File: index.html [hunk 6]
```diff
@@ -25,16 +25,31 @@
 </head>
 
 <body>
+
   <div class="container paper">
 
+    <div class="paper-navbar">
```
Comment (TotomInc) [inline index.html]: The goal of PaperCSS was to create the least classes possible, can you also add styling for `<nav>` tag? This way, people already using `<nav>` tag who include PaperCSS don't have to add any classes to their elements.

Comment (amanbayev) [reply index.html]: Ok I will rewrite it little later. Thank you for feedback!
Also, I will remove images in next push.

**Comment by rhyneav** [comment]
This is shaping up really nicely @amanbayev, thank you for all of the work you've put in so far!

@TotomInc has a great point. PaperCSS should be as minimal as possible in regards to adding classes to elements. Ideally, adding a `<nav>` should instantly make it look paper-y. We can always add classes to undo the paper-y look. This makes it easy to just drop in the PaperCSS files and have the page instantly look paper-ized. 

Also, I love the house svg @afzalsayed96!

**Comment by Kaperskyguru** [comment]
Wow nice ideals at paperCSS... I just learnt about it today through medium.... I want to contribute.... But am not good with CSS.... I can contribute in other ways right????

**Comment by TotomInc** [comment]
@Kaperskyguru you can contribute by giving ideas, creating issues if you encounter a bug, helping people with their issues, ...

**Comment by Kaperskyguru** [comment]
Ok.. Thanks appreciate.... @totominc

**Comment by amanbayev** [comment]
@afzalsayed96 I made flat version primary and removed button version.
@TotomInc I have removed junk from img folder
@rhyneav I have removed unneccesary classes

Thank you for great feedback! Navbar became super simple now. Easy to setup with least amount of customization. 
Because we're not using JS to operate mobile version, there must be a hidden checkbox input with label (icon), and that label should reference id of the checkbox input.

**Comment by amanbayev** [comment]
@afzalsayed96 could you help me with x-axis overflow issue? I couldn't find what is causing the problem.

**Comment by afzalsayed96** [comment]
Thanks @amanbayev ! I found out that the issue is in Tables section (reported in issue #52 ) and not caused due to navbar. Your commits are good to go 👍

**Comment by rhyneav** [comment]
Looking good, @amanbayev! For some reason, the navbar links are blue for me (Chrome). Is this happening for anyone else?

image

**Comment by afzalsayed96** [comment]
How strange! It's working normal for me on chrome as well as firefox

**Comment by afzalsayed96** [comment]
@amanbayev what happened to the navbar version with home icon on left?

Perhaps we need a back to top icon as the HTML page is pretty long and it is a pain to scroll back to top maybe at the bottom right

**Comment by amanbayev** [comment]
@afzalsayed96 I removed it :) I think having a separate icon-set and the allowing coders to pick what they need is better than hard-coding it.

I will contribute again to this PR later, sorry, I have been busy with work lately. Hoping to do a better commit after weekends.

**Comment by amanbayev** [comment]
@rhyneav I will look into this issue and perhaps include colors in my less/css file. Sorry I have been away 2 days. Need a few more days to finish off work. Any help from community to this PR will be appreciated however.

File: src/navbar.less [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+			position: static;
+			&:not(:first-child) {
+				display: none;
+			}
+			li {
+				&:not(:first-child) {
+					margin-top: 30px;
+				}
+				float: none;
+	      display: block;
```
Comment (TotomInc) [inline src/navbar.less]: Can you please properly indent the lines?

Comment (amanbayev) [reply src/navbar.less]: it must be Atom or something. I'm using 2 spaces in my editor preferences to indent

Comment (TotomInc) [reply src/navbar.less]: Hey, I recently added an `.editorconfig` and I realy recommend you to install (https://github.com/sindresorhus/atom-editorconfig)[this package for Atom]. Please, we need to keep things consistent across everyone.

**Comment by rhyneav** [comment]
Closing since #145 adds a navbar.

**Pull request closed without merging**